cns/fsnotify/fsnotify.go

```diff
@@ -164,8 +166,12 @@ func (w *watcher) readFS() error {
 // WatchFS starts the filesystem watcher to handle async Pod deletes.
 // Blocks until the context is closed; returns underlying fsnotify errors
 // if something goes fatally wrong.
-func (w *watcher) Start(ctx context.Context) error {
+func (w *watcher) Start(c context.Context) error {
+	ctx, cancel := context.WithCancel(c)
+	defer cancel()
+
 	errs := make(chan error)
```

Comment:
probably need to buffer this channel to 2. even if the context is scoped to here and `watchPendingDelete` and `watchFS` respect it, either/both may return something (could even be `nil` error) after `Start` returns, but they won't be able to send since nothing will be consuming and the goroutine leaks.

Reply:
what do you think about the update to errgroup pattern?